src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Migrations/20240501113006_InitialMigration.cs

```diff
@@ -579,40 +580,67 @@ protected override void Up(MigrationBuilder migrationBuilder)
+            var dataProcessorFunctionAppRoleName = isLocalEnvironment
+                ? "app_public_data_processor" 
+                : Environment.GetEnvironmentVariable("DataProcessorFunctionAppIdentityName");
+            if (dataProcessorFunctionAppRoleName != null)
             {
                 migrationBuilder.Sql(
-                    $"GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO \"{dataProcessorFunctionAppIdentityName}\"");
+                    $"""GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO {dataProcessorFunctionAppRoleName}""");
                 migrationBuilder.Sql(
-                    $"ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT, INSERT, UPDATE, DELETE ON TABLES TO \"{dataProcessorFunctionAppIdentityName}\"");
+                    $"""ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT, INSERT, UPDATE, DELETE ON TABLES TO {dataProcessorFunctionAppRoleName}""");
 
                 migrationBuilder.Sql(
-                    $"GRANT SELECT, UPDATE ON ALL SEQUENCES IN SCHEMA public TO \"{dataProcessorFunctionAppIdentityName}\"");
+                    $"""GRANT SELECT, UPDATE ON ALL SEQUENCES IN SCHEMA public TO {dataProcessorFunctionAppRoleName}""");
                 migrationBuilder.Sql(
-                    $"ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT, UPDATE ON SEQUENCES TO \"{dataProcessorFunctionAppIdentityName}\"");
+                    $"""ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT, UPDATE ON SEQUENCES TO {dataProcessorFunctionAppRoleName}""");
             }
 
             // Grant privileges on objects created by this resource's database user to the Publisher Function App user.
-            var publisherFunctionAppIdentityName = Environment.GetEnvironmentVariable("PublisherFunctionAppIdentityName");
-            if (publisherFunctionAppIdentityName != null)
+            var publisherFunctionAppRoleName = isLocalEnvironment
+                ? "app_publisher" 
+                : Environment.GetEnvironmentVariable("PublisherFunctionAppIdentityName");
+            if (publisherFunctionAppRoleName != null)
             {
                 migrationBuilder.Sql(
-                    $"GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO \"{publisherFunctionAppIdentityName}\"");
+                    $"""GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO {publisherFunctionAppRoleName}""");
                 migrationBuilder.Sql(
-                    $"ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT, INSERT, UPDATE, DELETE ON TABLES TO \"{publisherFunctionAppIdentityName}\"");
+                    $"""ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT, INSERT, UPDATE, DELETE ON TABLES TO {publisherFunctionAppRoleName}""");
             }
+
+            // Grant privileges on objects created by this resource's database user to the 'public_data_admin' role.
```

Comment:
For simplicity, would it be worth just having this role in local as well as Azure?  That way, if it needs tweaking in any way, it can be maintained and tested locally before changing in Azure.

Reply:
Previous comment referred to before I replaced `public_data_admin` with `public_data_read_write`, but the same is still true.

If you want to test it locally,  create a user like so

```bash
 psql -U postgres -d public_data <<-EOSQL
    CREATE ROLE my_user WITH LOGIN PASSWORD 'password';
    GRANT USAGE ON SCHEMA public TO my_user;
    GRANT public_data_read_write TO my_user;
EOSQL
```

I did this after deleting the public data docker volume and the `public_data.sql` init script in the data directory, started the database and then allowed the Public API app to migrate the database.

Logging in with this new user we see it has write privileges, for example it can insert data:

```bash
psql -U my_user -d public_data <<-EOSQL
    INSERT INTO "DataSets" ("Id", "Title", "Summary", "PublicationId", "Status", "Created")
    VALUES (gen_random_uuid(),'Title','Summary','cbbd299f-8297-44bc-92ac-558bcf51f8ad','Published',now());
EOSQL
```

Reply:
Cool